src/main/java/org/folio/rest/impl/LoginAPI.java

```diff
@@ -199,31 +232,218 @@ private Future<String> fetchToken(JsonObject payload, String tenant,
+
     request.putHeader(XOkapiHeaders.TENANT, tenant)
-      .putHeader(XOkapiHeaders.TOKEN, requestToken);
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
 
-    JsonObject payload = new JsonObject().put("userId", userId).put("sub", sub);
-    return request
-      .expect(ResponsePredicate.SC_CREATED)
-      .expect(ResponsePredicate.JSON)
-      .sendJsonObject(payload)
-      .map(response -> response.bodyAsJsonObject().getString("refreshToken")
-      );
+      return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private Future<HttpResponse<Buffer>> logoutAll(String tenant,
+      String okapiURL, String accessToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_LOGOUT_ALL_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.send();
+  }
+
+  private Future<HttpResponse<Buffer>> fetchRefreshToken(String tenant,
+      String okapiURL, String accessToken, String refreshToken) {
+    HttpRequest<Buffer> request = WebClientFactory.getWebClient(vertx).postAbs(okapiURL + TOKEN_REFRESH_ENDPOINT);
+
+    request.putHeader(XOkapiHeaders.TENANT, tenant)
+      .putHeader(XOkapiHeaders.TOKEN, accessToken);
+
+    return request.sendJson(new JsonObject().put(REFRESH_TOKEN, refreshToken));
+  }
+
+  private boolean usesTokenSignLegacy(String endpoint) {
+    return endpoint.equals(TOKEN_SIGN_ENDPOINT_LEGACY);
+  }
+
+
+  // For documentation of the cookie attributes please see:
+  // https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie
+  private String refreshTokenCookie(String refreshToken, String okapiUrl, String refreshTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(refreshTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var domain = URI.create(okapiUrl).getHost();
+    return Cookie.cookie(REFRESH_TOKEN, refreshToken)
+        .setMaxAge(ttlSeconds)
+        .setSecure(true)
+        .setPath("/authn")
+        .setHttpOnly(true)
+        .setDomain(domain)
+        .setSameSite(CookieSameSite.NONE)
+        .encode();
+  }
+
+  private String accessTokenCookie(String accessToken, String okapiUrl, String accessTokenExpiration) {
+    // The refresh token expiration is the time after which the token will be considered expired.
+    var exp = Instant.parse(accessTokenExpiration).getEpochSecond();
+    var ttlSeconds = exp - Instant.now().getEpochSecond();
+    var domain = URI.create(okapiUrl).getHost();
+    return Cookie.cookie(ACCESS_TOKEN, accessToken)
+      .setMaxAge(ttlSeconds)
+      .setSecure(true)
+      .setHttpOnly(true)
+      .setDomain(domain)
+      .setSameSite(CookieSameSite.NONE)
```

Comment:
Why is NONE needed?
mod-login-saml requires CookieSameSite.LAX, but LAX is the default.
Can setSameSite been removed?

Reply:
Tenants that don't use mod-login-saml should be able to use STRICT for highest security.
Can we make it configurable? For example
GET /authn/cookiesamesite
PUT /authn/cookiesamesite
defaulting to STRICT.

Reply:
> Why is NONE needed?

See my comment above.

Reply:
@julianladisch finally had a chance to test this and thought you would be interested. `SameSite=None` is the only option we have since we're doing CORS in our SPA. I tried `Lax` and `Strict` in my local test setup and the browser won't set the cookie if either one is used. Here's what FF says:

<img width="851" alt="image" src="https://user-images.githubusercontent.com/104455/185760034-3e7d7cf3-5a4e-4161-8d5b-0242dbd83b11.png">

Reply:
Some installations use the same host and port for front-end and back-end following https://github.com/folio-org/folio-ansible/blob/master/doc/index.md#replace-port-9130
They don't use CORS and want the most secure same site policy.
Example: https://folio-demo.gbv.de/

Reply:
I see it's proxying to https://folio-demo.gbv.de/okapi/ which would get around the need for OPTION requests and CORS. FYI looks like the cert is expired on that domain.

> Can we make it configurable? For example
> GET /authn/cookiesamesite

So I think it could be configurable. But I'm not sure I'm understanding what you're suggesting for the configuration. Would it be handled by separate endpoints which each return cookies with different `SameSite` attributes? Or would it be a system flag that is set when mod-login starts?

Seems like making it a configuration option (system flag) of mod-login would be better.

Reply:
I made the `SameSite=Lax` an option that can be configured through a system property or environment variable. Updated the readme about this.

I also did more testing without localhost. Added two other hosts to `etc/hosts` that map to localhost. The browser sees these not as localhost.

If I serve both my tester page and okapi from the same domain then `SameSite=Lax` is permitted.

If serve my tester page from a different domain than okapi (neither being localhost) with `SameSite=Lax` the cookies are rejected. If I set `SameSite=None` as noted above the cookies are accepted with 2 different hosts.